test/server.test.js

```diff
@@ -674,6 +674,42 @@ test('GH-77 uncaughtException (default behavior)', function(t) {
     });
 });
 
+// eslint-disable-next-line
+test('handleUncaughtExceptions should not call handler for internal errors', function(t) {
+    SERVER.get('/', function(req, res, next) {
```

Comment:
I'm a little bit confused about this test setup. I think we trying to verify restify/router errors (e.g., 404) don't trigger uncaught exceptions? If yes, should we just remove the get route and have the client do a get against a non-existent route?

Reply:
There seems to be a behavior difference if you have no routes (which 404s) and having only GET routes and trying HEAD. I agree that this makes the test confusing, but I think if the behavior difference should be fixed, it should be fixed separately.

Reply:
Gotcha - thanks for context! I expect this is related to CORS use cases - do you mind adding all of this as a comment to the test? Otherwise, it's a bit confusing to register the empty `/` that we actually don't expect to be run or triggered. Maybe we could do an `assert.fail('should not run')` or something inside the GET handler for clarity as well?

Reply:
Makes sense. I added a comment and the assertion that you mentioned.